server/game/GameActions/ClaimFavorAction.ts

```diff
@@ -0,0 +1,36 @@
+import { PlayerAction, PlayerActionProperties} from './PlayerAction';
+import { EventNames } from '../Constants';
+import AbilityContext = require('../AbilityContext');
+import Player = require('../player');
+import { GameAction, GameActionProperties } from './GameAction';
+
+export interface ClaimFavorProperties extends PlayerActionProperties {
+    target: Player | null;
+}
+
+export class ClaimFavorAction extends PlayerAction {
+    name = 'discardFavor';
+    eventName = EventNames.OnClaimFavor;
+    effect = '{0} succesfully claims the Emperor\'s favor.';
```

Comment:
this `effect` is similar to `effect` in a triggered ability - it should be `'claim the Emperor\'s favor'`